### ISSUE TITLE
Popover: Shift-tab never closes popover without portal

### ISSUE THREAD

**Issue by kfranqueiro** [chakra-ui/chakra-ui#4385]
### Description

When I shift-tab from the container or the first tabbable child of the popover content, I expect focus to return to the trigger and the popover to close, but it never closes

### Link to Reproduction

https://codesandbox.io/s/zen-jang-ueh31?file=/src/index.tsx

### Steps to reproduce

1. Go to https://codesandbox.io/s/zen-jang-ueh31?file=/src/index.tsx
2. Click on "Open popover"
3. Press shift-tab
  * Focus returns to the trigger button; popover remains open
4. Press shift-tab again
  * Focus continues backwards in the document; popover still remains open

Alternatively:

1. Go to https://codesandbox.io/s/zen-jang-ueh31?file=/src/index.tsx
2. Click on "Open popover"
3. Press tab to move focus to the first tabbable element inside the popover (the close button)
4. Press shift-tab
  * Focus returns to the trigger button, then immediately "bounces" back into the open container

TL;DR `closeOnBlur` never actually closes on blur in the backwards direction for a popover without a portal.

### Chakra UI Version

Reproduced on latest and also on 1.3.3 (popover 1.2.0)

### Browser

Google Chrome 93

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

The code [here](https://github.com/chakra-ui/chakra-ui/blob/3205ad6d55f935f37db8f61d2dde7eaa2481578c/packages/popover/src/use-popover.ts#L225-L226) seems to intentionally make an exception for focus moving to the trigger element. I am wondering whether this exception should only be made if the trigger mode is hover, not click?

I'm guessing this doesn't typically happen in the case where a portal is used because the blur event doesn't see the focus moving directly back to the trigger element.

**Comment by kfranqueiro** [comment]
This is still an issue with latest chakra using the reproduction steps + codesandbox.

**Comment by kfranqueiro** [comment]
This is still an issue in the latest version of Chakra.

**Comment by kfranqueiro** [comment]
I'm not sure why the bot closed this when I commented after its last comment?
